crates/storage/db-api/src/models/metadata.rs

```diff
@@ -0,0 +1,31 @@
+//! Storage metadata models.
+
+use reth_codecs::{add_arbitrary_tests, Compact};
+use serde::{Deserialize, Serialize};
+
+/// Storage configuration settings for this node.
+///
+/// These should be set during `init_genesis` or `init_db` depending on whether we want dictate
+/// behaviour of new or old nodes respectively.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default, Serialize, Deserialize, Compact)]
+#[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
+#[add_arbitrary_tests(compact)]
+pub struct StorageSettings {
+    /// Whether this node always writes receipts to static files.
+    ///
+    /// If this is set to FALSE AND receipt pruning IS ENABLED, all receipts should be written to DB. Otherwise, they should be written to static files. This ensures that older nodes do not need to migrate their current DB tables to static files. For more, read: <https://github.com/paradigmxyz/reth/issues/18890#issuecomment-3457760097>
+    pub receipts_on_static_files: bool,
+}
```

Comment:
can we make this so that we can easily extend this and store arbitrary info?

Reply:
https://github.com/paradigmxyz/reth/blob/642925011d6ccfe5aff18896f7eb279fd0ffb76e/crates/storage/storage-api/src/metadata.rs#L35-L39

this is also being encoded as `serde_json` so extending this should be fine